doc/index.rst

```diff
@@ -0,0 +1,366 @@
+used with a Python session and Firefly server that are both remote and
+situated close to the data.
+
+A user typically will not import `lsst.display.firefly` directly. Instead,
+the `lsst.afw.display` interface will commonly be used with `backend=firefly`.
+
+.. _lsst-display-firefly-getting-started:
+
+Getting Started
+===============
+
+This short example uses data that has been run through the
+`LSST stack demo <https://pipelines.lsst.io/install/demo.html>`_.
+
+
+At a shell prompt:
+
+.. code-block:: shell
+   :name: gs-shell-setup
+
+   setup obs_sdss
+   setup display_firefly
+
+
+Start a Python session of some kind in the directory where the demo was run.
+Then these statements will create a Butler instance for the output data
+and then retrieve an exposure.
+
+.. code-block:: py
+   :name: gs-butler-get
+
+   from lsst.daf.persistence import Butler
+   butler = Butler('./output')
+   calexp = butler.get('calexp', run=4192, camcol=4, field=300, filter='g')
+
+The next lines will set up the display, using a public Firefly server at IPAC.
+
+.. code-block:: py
+   :name: gs-setup-display
+
+   import lsst.afw.display as afw_display
+   display1 = afw_display.Display(frame=1, backend='firefly',
+      host='irsa.ipac.caltech.edu', port=80, basedir='irsaviewer',
+      channel='mychannel')
+
+Open a browser window to
+`http://irsa.ipac.caltech.edu/irsaviewer/;wsch=mychannel <http://irsa.ipac.caltech.edu/irsaviewer/;wsch=mychannel>`_.
+Then display the exposure:
+
+.. code-block:: py
+   :name: gs-display-calexp
+
+   display1.mtv(calexp)
+
+The displayed exposure will appear in your browser.
+
+.. _lsst-display-firefly-using:
+
+Using lsst.display.firefly
+==========================
+
+The `lsst.display.firefly` package is not used directly. Instead, instances
+of `lsst.afw.display` are created with parameters passed to the backend.
+The Firefly backend is based upon the Python API in
+:class:`firefly_client.FireflyClient`.
+
+Setup
+-----
+
+Before a Python session or Jupyter notebook is started, setup of stack
+packages must be completed. Use :command:`setup display_firefly` to enable
+the Firefly backend.
+
+Initializing with lsst.afw.display
+----------------------------------
+
+The recommended way to create a display object for Firefly is using
+the :class:`Display` constructor from `lsst.afw.display`:
+
+.. code-block:: py
+    :name: construct-display
+
+    import lsst.afw.display as afw_display
+    display1 = afw_display.Display(frame=1, host='localhost', port=8080,
+                                   basedir='firefly', channel='afw')
+
+The parameters shown above (besides ``frame``) are the defaults and will
+apply when running a Firefly server locally with default settings.
+
+If a Firefly server has been provided to you, set ``host``, ``port``, and
+``basedir`` according to the informatio provided. You should set ``channel``
+to a unique string to avoid another user from writing to your display.
+
+.. warning::
```

Comment:
I'm not sure I understand this warning.  Are you saying an instance of `Display` has no mechanism for changing the endpoint, so if you want to change the endpoint, you have to create a new instance?

Or are you saying that within an interpreter session you can only have on endpoint over that session?

Reply:
In an interpreter session, you can have only one endpoint. This is because a the underlying `FireflyClient` is a global variable within `lsst.display.firefly`. It would be good to allow this to be overridden but I am not sure how to implement that.